### PR TITLE
Support any number of networks greater than 0

### PR DESCRIPTION
**Pull request by KillariDev**

### CODE REVIEW

File: app/ts/constants.ts [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+		blockExplorer: 'https://sepolia-etherscan.io/',
+		rpcUrl: 'https://rpc-sepolia.dark.florist/flipcardtrustone',
+		chainId: `0x${11155111n.toString(16)}`,
+		blockExplorerApi: 'https://sepolia-api.etherscan.io'
+	}]*/
+])
+
+export const findNetworkBySimulationRelayEndpoint = (simulationRelayEndpoint: string) => {
+	const networks = Array.from(NETWORKS, ([chainid, network]) => ({ chainid, network }))
+	return networks.find((network) => network.network.simulationRelay === simulationRelayEndpoint)?.network
```
Comment (jubalm) [inline app/ts/constants.ts:26]: `NETWORKS` should already be iterable

```suggestion
	for (const [_chainId, network] of NETWORKS) {
		if (network.simulationRelay === simulationRelayEndpoint) return network
	}
	return undefined
```

File: app/ts/constants.ts [hunk 2]
```diff
@@ -1,3 +1,27 @@
-export const NETWORKS: { [chainId: string]: { simulationRelay: string, submissionRelay: string, blockExplorer: string, rpcUrl: string }} = {
-	'1': { simulationRelay: 'https://flashbots-cors-proxy.dark-florist.workers.dev/', submissionRelay: 'https://rpc.titanbuilder.xyz', blockExplorer: 'https://etherscan.io/', rpcUrl: 'https://rpc.dark.florist/flipcardtrustone' },
+type Network = { simulationRelay: string, submissionRelay: string, blockExplorer: string, rpcUrl: string, chainId: string, blockExplorerApi: string }
+
+export const MAINNET = {
```
Comment (jubalm) [inline app/ts/constants.ts:3]: just to be explicit with type
```suggestion
export const MAINNET: Network = {
```

**Review by jubalm — APPROVED** [review]

File: app/ts/constants.ts (hunk 2 above)
Comment (KillariDev) [reply app/ts/constants.ts:3]: the mapping below would already fail if this doesn't match?

File: app/ts/constants.ts [hunk 3]
```diff
@@ -1,3 +1,35 @@
...
+	networkName: 'Mainnet',
+	simulationRelay: 'https://flashbots-cors-proxy.dark-florist.workers.dev/',
+	submissionRelay: 'https://rpc.titanbuilder.xyz',
+	blockExplorer: 'https://etherscan.io/',
+	rpcUrl: 'https://rpc.dark.florist/flipcardtrustone',
+	chainId: `0x${1n.toString(16)}`,
+	blockExplorerApi: 'https://api.etherscan.io'
+}
```
Comment (MicahZoltu) [inline app/ts/constants.ts:11]: Why not define this inline and then reference it with `NETWORKS[1n]` when you need it?

Alternatively, we could do something like:
```ts
export const MAINNET_ID = 1n
export const SEPOLIA_ID = 11155111n
```
Then you could access it with `NETWORKS[MAINNET_ID]`.

Could also do an enum if you wanted, though I'm not a huge fan in TS.

File: app/ts/constants.ts [hunk 4]
```diff
@@ -1,3 +1,35 @@
...
+
+export const NETWORKS = new Map<bigint, Network>([
```
Comment (MicahZoltu) [inline app/ts/constants.ts:13]: Why a `Map` instead of an object?

**Review by MicahZoltu — APPROVED** [review]

File: app/ts/constants.ts (hunk 4 above)
Comment (KillariDev) [reply app/ts/constants.ts:13]: I want to be able to access networks that don't exist (and find out they are undefined)

File: app/ts/constants.ts (hunk 3 above)
Comment (KillariDev) [reply app/ts/constants.ts:11]: I don't need other networks like that, just the default mainnet one. I guess make an object out of it, and then convert that object to a map, but that sounds more complex

File: app/ts/constants.ts (hunk 4 above)
Comment (KillariDev) [reply app/ts/constants.ts:13]: I could use funtypes parsing to achieve that too tho. I am not sure if its any better

Comment (MicahZoltu) [reply app/ts/constants.ts:13]: You can access networks that don't exist on an object too.
```ts
const NETWORKS: { [bigint]: Network } = { ... }
NETWORKS[42] // undefined
NETWORKS[1] // { ... }
```

**Pull request merged**